docs/Dialects/FIRRTL/FIRRTLAnnotations.md

```diff
@@ -1458,6 +1458,66 @@ Example:
 }
 ```
 
+### OutputDirAnnotation
+
+| Property   | Type   | Description                             |
+| ---------- | ------ | --------------------------------------- |
+| class      | string | `circt.OutputDirAnnotation`             |
+| dirname    | string | The output directory                    |
+| target     | string | Reference target                        |
+
+Specify the output directory for a module. The target must be a module, and must
+be local.
+
+Example:
+```json
+{
+  "class": "circt.OutputDirAnnotation",
+  "dirname": "verification",
+  "target": "~Foo|Bar"
+}
+```
+
+### OutputDirPrecedenceAnnotation
+
+| Property   | Type    | Description                             |
+| ---------- | ------- | --------------------------------------- |
+| class      | string  | `circt.OutputDirPrecedenceAnnotation`   |
+| name       | string  | The output directory                    |
+| parent     | string  | The parent output directory             |
+
+Specify the "parent" of an output directory.
+
+When Verilog is output, some modules will have user-specified output
+directories. When a module `M` is only instantiated by modules which have a
+common output directory `D`, we can also output module `M` in that same
+directory `D`.
```

Comment:
Why are we not just putting it where the OutputDirAnnotation says?

Reply:
If there is an OutputDirAnnotation, we will use that :+1:. We only assign output directories for modules that are:
- private
- don't already have user-specified output directories